distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
@anusha-ragunathan to me it more about separation of the code. This code should not be pulling plugins, and it really doesn't even need to know about anything that is a non-image. Having the whitelist would also negate the need of having the `v0` version defined in distribution since the code change can be isolated to the image pull logic.